app/models/task.py

```diff
@@ -1,5 +1,47 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from sqlalchemy import ForeignKey
 from ..db import db
+from datetime import datetime
+from typing import Optional
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    description: Mapped[str]
+    completed_at: Mapped[Optional[str]]
+    goal_id: Mapped[Optional[int]] = mapped_column(ForeignKey("goal.id"))
+    goal: Mapped[Optional["Goal"]] = relationship(back_populates="tasks")
+
+
+    # from model to JSON
+    def to_dict(self):
+        if self.goal_id is None:
+            return dict(
+                    id=self.id,
+                    title=self.title,
+                    description=self.description,
+                    is_complete=self.completed_at!=None
+                    )
+        else:
+            return dict(
+                    id=self.id,
+                    title=self.title,
+                    description=self.description,
+                    is_complete=self.completed_at!=None,
+                    goal_id=self.goal_id
+                    )
```

Comment:
Nice. This addresses the underlying test issue. Notice that there is a significant amount of repetition between the two halves of the conditional. Instead of duplicating so much of the dictionary creation, we could take an approach more like this. (Also remember that we prefer `is`/`is not` to compare with `None`)
```py
        result = dict(
            id=self.id,
            title=self.title,
            description=self.description,
            is_complete=self.completed_at is not None
            )

        if self.goal_id:
            result["goal_id"] = self.goal_id

        return result
```